src/CompositeNormL1.jl

```diff
@@ -0,0 +1,48 @@
+# Composition of the L1 norm with a function
+export CompositeNormL1
+
+@doc raw"""
+    CompositeNormL1(λ, c!, J!, A, b)
+
+Returns the ``\ell_{1}`` norm operator composed with a function
+```math
+f(x) = λ \|c(x)\|_1
+```
+where ``\lambda \geq 0``. c! and J! should be functions
+```math
+\begin{aligned}
+&c(x) : \mathbb{R}^n \xrightarrow[]{} \mathbb{R}^m \\
+&J(x) : \mathbb{R}^n \xrightarrow[]{} \mathbb{R}^{m\times n}
+\end{aligned}
+```
+such that J is the Jacobian of c. A and b should respectively be a matrix and a vector which can respectively store the values of J and c.
```

Comment:
It would be useful to state how `c!()` and `J!()` are expected to be called and what they should return. In particular, is `A` expected to be dense or sparse? In NLPModels, there is no Jacobian method that fills in a matrix, i.e., something like `jac!(nlp, x, A)` does not exist. However, there is `jac_coord!(nlp, x, vals)`, which fills in the array `vals` of nonzeros elements in the sparse coordinate representation of the Jacobian.

Reply:
I added a commit, see MaxenceGollier:affineNormL1@561c415e735065a7a94356e6476f898f3a2d49d8
I just changed the documentation, do you expect to force A by changing the type in the constructor ? I don't know if it is necessary.